text_extensions_for_pandas/array/test_tensor.py

```diff
@@ -207,6 +207,25 @@ def test_slice(self):
         expected = np.array([[3, 4], [5, 6]])
         npt.assert_array_equal(expected, result)
 
+    def test_bool_indexing(self):
+        s = TensorArray([[1, 2], [3, 4]])
+
+        result = s[[True, True]]
+        expected = np.array([[1, 2], [3, 4]])
+        npt.assert_array_equal(result, expected)
+
+        result = s[[True, False]]
+        expected = np.array([[1, 2]])
+        npt.assert_array_equal(result, expected)
+
+        result = s[[False, True]]
+        expected = np.array([[3, 4]])
+        npt.assert_array_equal(result, expected)
+
+        result = s[[False, False]]
+        expected = np.empty((0, 2))
+        npt.assert_array_equal(result, expected)
+
```

Comment:
You need to test whether indexing with Boolean masks into `pd.Series` of `TensorArray` works.

You also need to test whether indexing with a Boolean mask into a DataFrame works. See the example code in #16

Reply:
ah ok. It appeared to be the same issue when I first looked, but best to verify all cases.

Reply:
All of these expressions need to return the same type, both for consistency with the way Pandas handles selections with Boolean masks; and to avoid causing headaches for user code that computes a Boolean mask in a data-dependent manner. The code:
``` python
my_df[colname][<expression that returns a Boolean mask>]
```
should always evaluate an object of the same type; otherwise later code will receive unexpected types.